src/EditorFeatures/Core/FindUsages/ClassifiedSpansAndHighlightSpanFactory.cs

```diff
@@ -44,6 +49,38 @@ internal static class ClassifiedSpansAndHighlightSpanFactory
                 documentSpan.Document, documentSpan.SourceSpan, cancellationToken).ConfigureAwait(false);
         }
 
+        public static ClassifiedTextElement BuildClassifiedTextElementForClassifiedSpans(IEnumerable<ClassifiedSpan> classifiedSpans, ITextSnapshot snapshot, CancellationToken cancellationToken)
+        {
+            Contract.Assert(classifiedSpans != null && classifiedSpans.Any());
+
+            // Convert spans to textruns
+            var runs = new List<ClassifiedTextRun>();
+            var lastSpanEnd = -1;
+            foreach (var span in classifiedSpans)
+            {
+                // Add whitespace
+                if (lastSpanEnd > 0 && span.TextSpan.Start > lastSpanEnd)
+                {
+                    // find the gap between last span and current span, read the gap text from document
+                    var spanGap = new Span(lastSpanEnd, span.TextSpan.Start - lastSpanEnd);
+                    var spanText = snapshot.GetText(spanGap);
+                    runs.Add(new ClassifiedTextRun(ClassificationTypeNames.WhiteSpace, RemoveIndentsAfterNewLine(spanText)));
+                }
+
+                lastSpanEnd = span.TextSpan.End;
+                runs.Add(new ClassifiedTextRun(span.ClassificationType, snapshot.GetText(span.TextSpan.ToSpan())));
+            }
+
+            return new ClassifiedTextElement(runs);
+        }
```

Comment:
I think we should break this function into a few functions.  

Function one goes from ```ImmutableArray<ClassifiedSpan>``` to ```ImmutableArray<ClassifiedSpan>```.  It's only purpose is to fill in the gaps of the original sequence of classified spans.

Function two ```ImmutableArray<ClassifiedSpan>``` to ```ImmutableArray<ClassifiedSpan>```.  It's only purpose is to trim leading *common* indentation from each line.

Function three finally takes all these spans, and makes the classified runs out of them.  

When you have this, it will then be easy to write unit tests that validate the results of the first two functions.  i.e. we should be able to say (in a unit test), if you're getting quick info on this ```}``` thne we expect this exact sequence of classified spans.